### ISSUE TITLE
Remove not used code regarding MyData

### ISSUE THREAD

**Issue by madryk** [CeON/fairchive#2081]
At start I would go to:
`MyDataPage` and remove it with corresponding  `xhtml` where it have reference. Also
I would look if `MyDataPage` is the only place when we invoke some methods (for example:
`MyDataFilterParams.getPublishedStatesForMyDataPage`)
and remove them also (go as deep as possible).
Next place is: `src/main/webapp/mydata_templates` - it shouldn't have any references in the app
but we could check it to be sure.
Next maybe go to `DataRetrieverAPI` - remove unused methods.
Next I would go to `MyDataFilterParams` and do standard check for unused fields, methods,
maybe switching public -> private methods where possible.
Basically check every class in `edu.harvard.iq.dataverse.mydata` and do that kind of checks
